File: Tyche.iOS/Bet/Sources/Bet/Domain/PoolGamblerBet.swift
import Foundation
import Core

public struct PoolGamblerBet: Codable {
    let poolId: String
    let gamblerId: String
    let matchId: String
    let homeTeamId: String
    let homeTeamName: String
    let awayTeamId: String
    let awayTeamName: String
    let matchScore: TeamScore<Int>?
    let betScore: TeamScore<Int>?
    let score: Int?
    let matchDateTime: Date
    let isLocked: Bool
}

extension PoolGamblerBet {
    func toPoolGamblerBetModel() -> PoolGamblerBetModel {
        return PoolGamblerBetModel(
            poolId: self.poolId,
            gamblerId: self.gamblerId,
            matchId: self.matchId,
            homeTeamId: self.homeTeamId,
            homeTeamName: self.homeTeamName,
            awayTeamId: self.awayTeamId,
            awayTeamName: self.awayTeamName,
            matchScore: self.matchScore,
            betScore: self.betScore,
            score: self.score,
            matchDateTime: self.matchDateTime,
            isLocked: self.isLocked
        )
    }
}

File: Amazon.Lambda/Felipearpa.Tyche.Pool/src/Domain/PoolGamblerBet.fs
namespace Felipearpa.Tyche.Pool.Domain

open System
open Felipearpa.Tyche.Pool.Type
open Felipearpa.Type

type PoolGamblerBet =
    { PoolId: Ulid
      GamblerId: Ulid
      GamblerUsername: NonEmptyString100
      MatchId: Ulid
      PoolLayoutId: Ulid
      HomeTeamId: string
      HomeTeamName: NonEmptyString100
      AwayTeamId: string
      AwayTeamName: NonEmptyString100
      MatchScore: TeamScore<int> option
      BetScore: TeamScore<BetScore> option
      Score: int option
      MatchDateTime: DateTime
      Round: string
      GroupName: string option
      isLocked: bool
      isComputed: bool }
